unit/src/MooseServerTest.C

```diff
@@ -663,7 +1141,164 @@ TEST_F(MooseServerTest, DocumentCompletionRequest)
+                                                   response_errors,
+                                                   response_id,
+                                                   response_is_incomplete,
+                                                   completions_array));
+
+  EXPECT_TRUE(response_errors.str().empty());
+
+  EXPECT_EQ(request_id, response_id);
+
+  EXPECT_EQ(5, (int)completions_array.size());
+
+  std::ostringstream completions_actual;
+
+  format_completions(completions_array, completions_actual);
+
+  // expected completions with zero-based lines and columns
+
+  std::string completions_expect = R"INPUT(
+label: Eigenvalue         text: Eigenvalue         desc: Eigenvalue solves a standard/generalized linear... pos: [30.9]-[30.18]
+label: InversePowerMethod text: InversePowerMethod desc: Inverse power method for eigenvalue problems.      pos: [30.9]-[30.18]
+label: NonlinearEigen     text: NonlinearEigen     desc: Executioner for eigenvalue problems.               pos: [30.9]-[30.18]
+label: Steady             text: Steady             desc: Executioner for steady-state simulations.          pos: [30.9]-[30.18]
+label: Transient          text: Transient          desc: Executioner for time varying simulations.          pos: [30.9]-[30.18]
+)INPUT";
+
+  EXPECT_EQ(completions_expect, "\n" + completions_actual.str());
+}
+
+TEST_F(MooseServerTest, CompletionValueInputLookups)
+{
+  // completion test parameters - on displacements parameter value in VacuumBC
+
+  int request_id = 4;
+  std::string doc_uri = wasp::lsp::m_uri_prefix + std::string("/test/input/path");
+  int line = 26;
+  int character = 21;
+
+  // build completion request with the test parameters
+
+  wasp::DataObject completion_request;
+  std::stringstream completion_errors;
+
+  EXPECT_TRUE(wasp::lsp::buildCompletionRequest(
+      completion_request, completion_errors, request_id, doc_uri, line, character));
+
+  EXPECT_TRUE(completion_errors.str().empty());
+
+  // handle the built completion request with the moose_server
+
+  wasp::DataObject completion_response;
+
+  EXPECT_TRUE(moose_server->handleCompletionRequest(completion_request, completion_response));
+
+  EXPECT_TRUE(moose_server->getErrors().empty());
+
+  // check the dissected values of the moose_server completion response
+
+  std::stringstream response_errors;
+  int response_id;
+  bool response_is_incomplete;
+  wasp::DataArray completions_array;
+
+  EXPECT_TRUE(wasp::lsp::dissectCompletionResponse(completion_response,
+                                                   response_errors,
+                                                   response_id,
+                                                   response_is_incomplete,
+                                                   completions_array));
+
+  EXPECT_TRUE(response_errors.str().empty());
+
+  EXPECT_EQ(request_id, response_id);
+
+  EXPECT_EQ(4, (int)completions_array.size());
```

Comment:
```suggestion
  EXPECT_EQ(4u, completions_array.size());
```

Reply:
Thanks, I pushed a commit that address this.